includes/class-wcs-object-data-cache-manager.php

```diff
@@ -160,6 +170,13 @@ public function prepare_object_changes( $object, $generate_type = 'changes_only'
 				];
 			}
 		}
+
+		// If the object is being deleted, we want to record all the changes as deletes.
+		if ( $is_delete_object ) {
+			foreach ( $this->object_changes[ $object->get_id() ] as $data_key => $data ) {
+				$this->object_changes[ $object->get_id() ][ $data_key ]['type'] = 'delete';
```

Comment:
When for the delete `type` the `previous` value also needs to be set otherwise you get this error: 

```
PHP Warning:  Undefined array key "previous" in /woocommerce-subscriptions-core/includes/class-wcs-object-data-cache-manager.php on line 294
```
I fixed this locally by doing this: 

```suggestion
				$this->object_changes[ $object->get_id() ][ $data_key ]['type'] = 'delete';
				$this->object_changes[ $object->get_id() ][ $data_key ]['previous'] = $data['new'];
				unset( $this->object_changes[ $object->get_id() ][ $data_key ]['new'] );
```

However given the feedback above, this might need to be handled more broadly.

Reply:
Thank you for pointing that out. I have address this feedback.